daemon/restore.go

```diff
@@ -0,0 +1,59 @@
+package daemon
+
+import (
+	"fmt"
+	"path/filepath"
+
+	"github.com/docker/docker/container"
+	"github.com/docker/docker/runconfig"
+)
+
+// ContainerRestore restores the process in a container with CRIU
+func (daemon *Daemon) ContainerRestore(name string, opts *runconfig.CriuConfig, forceRestore bool) error {
+	container, err := daemon.Get(name)
+	if err != nil {
+		return err
+	}
+
+	if !forceRestore {
+		// TODO: It's possible we only want to bypass the checkpointed check,
+		// I'm not sure how this will work if the container is already running
+		if container.IsRunning() {
+			return fmt.Errorf("Container %s already running", name)
+		}
+
+		if !container.IsCheckpointed() {
+			return fmt.Errorf("Container %s is not checkpointed", name)
+		}
+	} else {
```

Comment:
Would it make sense to at least check the image ID match here? We could store it in `criu.work` for example. Image IDs are quite stable now after content-addressability.